bin/node/runtime/src/impls.rs

```diff
@@ -142,49 +197,83 @@ mod tests {
 			(0, fm.clone()),
 			(100, fm.clone()),
+			(1000, fm.clone()),
 			(target(), fm.clone()),
 			(max() / 2, fm.clone()),
 			(max(), fm.clone()),
 		];
 		test_set.into_iter().for_each(|(w, fm)| {
 			run_with_system_weight(w, || {
 				assert_eq_error_rate!(
-					fee_multiplier_update(w, fm),
-					TargetedFeeAdjustment::<TargetBlockFullness>::convert(fm),
-					// Error is only 1 in 10^18
-					Multiplier::from_inner(1),
+					truth_value_update(w, fm),
+					runtime_multiplier_update(fm),
+					// Error is only 1 in 100^18
+					Multiplier::from_inner(100),
 				);
 			})
 		})
 	}
 
 	#[test]
-	fn empty_chain_simulation() {
-		// just a few txs per_block.
-		let block_weight = 0;
-		run_with_system_weight(block_weight, || {
-			let mut fm = Multiplier::default();
+	fn multiplier_can_grow_from_zero() {
+		// if the min is too small, then this will not change, and we are doomed forever.
+		// the weight is 1/10th bigger than target.
+		run_with_system_weight(target() * 101 / 100, || {
+			let next = runtime_multiplier_update(MIN_MULTIPLIER);
+			assert!(next > MIN_MULTIPLIER, "{:?} !>= {:?}", next, MIN_MULTIPLIER);
+		})
+	}
+
+	#[test]
+	fn multiplier_cannot_go_below_limit() {
+		// will not go any further below even if block is empty.
+		run_with_system_weight(0, || {
+			let next = runtime_multiplier_update(MIN_MULTIPLIER);
+			assert_eq!(next, MIN_MULTIPLIER);
+		})
+	}
+
+	#[test]
+	fn time_to_reach_zero() {
+		// blocks per 24h in substrate-node: 28,800 (k)
+		// s* = 0.1875
+		// The bound from the research in an empty chain is:
+		// v <~ (p / k(0 - s*))
+		// p > v * k * -0.1875
+		// to get p == -1 we'd need
+		// -1 > 0.00001 * k * -0.1875
+		// 1 < 0.00001 * k * 0.1875
+		// 10^9 / 1875 < k
+		// k > 533_333 ~ 18,5 days.
+		run_with_system_weight(0, || {
+			// start from 1, the default.
+			let mut fm = Multiplier::one();
 			let mut iterations: u64 = 0;
 			loop {
-				let next = TargetedFeeAdjustment::<TargetBlockFullness>::convert(fm);
+				let next = runtime_multiplier_update(fm);
 				fm = next;
-				if fm == Multiplier::saturating_from_integer(-1) { break; }
+				if fm == MIN_MULTIPLIER { break; }
 				iterations += 1;
 			}
-			println!("iteration {}, new fm = {:?}. Weight fee is now zero", iterations, fm);
-			assert!(iterations > 50_000, "This assertion is just a warning; Don't panic. \
-				Current substrate/polkadot node are configured with a _slow adjusting fee_ \
-				mechanism. Hence, it is really unlikely that fees collapse to zero even on an \
-				empty chain in less than at least of couple of thousands of empty blocks. But this \
-				simulation indicates that fees collapsed to zero after {} almost-empty blocks. \
-				Check it",
-				iterations,
-			);
+			assert!(iterations > 533_333);
+		})
+	}
+
+	#[test]
+	fn max_change_per_day() {
+		run_with_system_weight(max(), || {
+			let mut fm = Multiplier::one();
+			// See the example in the module doc. Changes can be at most 0.216, hence fm < 1.216
```

Comment:
I think this is wrong, need to double check. Changes need to be at **least** `0.216`.